src/main/java/gov/usgs/cida/pubs/busservice/sipp/SippConversionService.java

```diff
@@ -0,0 +1,301 @@
+			LinkType linkType = new LinkType();
+			linkType.setId(LinkType.INDEX_PAGE);
+			publicationLink.setLinkType(linkType);
+			Collection<PublicationLink<?>> publicationLinks = Arrays.asList(publicationLink);
+			return publicationLinks;
+		} else {
+			return null;
+		}
+	}
+
+	protected PublishingServiceCenter buildPublishingServiceCenter(String text) {
+		if (null != text) {
+			PublishingServiceCenter publishingServiceCenter = new PublishingServiceCenter();
+			publishingServiceCenter.setText(text);
+			return publishingServiceCenter;
+		} else {
+			return null;
+		}
+	}
+
+	protected Integer buildRank(String rawRank) {
+		Integer rank = PubsUtils.parseInteger(rawRank);
+		if (null == rank) {
+			return 1;
+		} else {
+			return rank;
+		}
+	}
+
+	protected PublicationSeries buildSeriesTitle(PublicationSubtype pubSubtype, String text) {
+		if (null == pubSubtype || null == text) {
+			return null;
+		} else {
+			PublicationSeries publicationSeries = new PublicationSeries();
+			publicationSeries.setPublicationSubtype(pubSubtype);
+			publicationSeries.setText(text);
+			return publicationSeries;
+		}
+	}
+
+	protected String buildTitle(String finalTitle, String workingTitle) {
+		if (null == finalTitle) {
+			return workingTitle;
+		} else {
+			return finalTitle;
+		}
+	}
+
+	protected UsgsContributor buildUsgsContributor(Author author) {
+		UsgsContributor contributor = new UsgsContributor();
+//TODO - if added to SIPP...
+//		contributor.setFamily(parser.getFirstNodeText(doc.getDocumentElement(), Schema.LAST_NAME));
+//		contributor.setGiven(parser.getFirstNodeText(doc.getDocumentElement(), Schema.FIRST_NAME));
+
+		String[] familyGiven = author.splitFullName();
+		contributor.setFamily(familyGiven[0]);
+		contributor.setGiven(familyGiven[1]);
+
+		contributor.setOrcid(PubsUtils.normalizeOrcid(author.getOrcid()));
+		contributor.setPreferred(true);
+
+		if (null != author.getCostCenter()) {
+			CostCenter costCenter = new CostCenter();
+			costCenter.setText(author.getCostCenter());
+			contributor.getAffiliations().add(costCenter);
+		}
+
+		return contributor;
+	}
+
+	protected String cleanseUsgsSeriesNumber(String usgsSeriesNumber) {
+		if (null == usgsSeriesNumber
+				|| ".".contentEquals(usgsSeriesNumber)) {
+			return null;
+		} else {
+			return usgsSeriesNumber;
+		}
+	}
+
+	protected Collection<MpPublicationContributor> fixRanks(final List<MpPublicationContributor> contributors) {
+		Set<Integer> ranks = new HashSet<>();
+		for (Iterator<MpPublicationContributor> contributorsIter = contributors.iterator(); contributorsIter.hasNext();) {
+			MpPublicationContributor contributor = contributorsIter.next();
+			ranks.add(contributor.getRank());
+		}
+
+		if (contributors.size() != ranks.size()) {
+			Integer i = 0;
+			for (Iterator<MpPublicationContributor> fixIterator = contributors.iterator(); fixIterator.hasNext();) {
+				MpPublicationContributor fixMe = fixIterator.next();
+				i++;
+				fixMe.setRank(i);
```

Comment:
Is the order of contributors.iterator() known and guaranteed? If not, i might not be the expected value in
fixMe.setRank(i);